server/commands/connected_objects.go

```diff
@@ -167,22 +242,52 @@ func (c *ConnectedObjectsCommand) reverseExpandTupleToUserset(
 			continue
 		}
 
-		targetUserStr := req.targetUserRef.GetObject()
+		user := t.GetUser()
```

Comment:
line 245-258 is rather similar to line 398-411?  Will it make sense to move to a function called constructTargetUserStr (or something like that)?

Reply:
Honestly, I introduced the `isUserRef` interface to distinguish between different types of users, but I'm not sure I like the changes. I am seriously considering going back to the `ObjectRelation` approach (like the code we had before), because it makes the code more uniform.

I'm debating what direction I want to go...

Reply:
Similar comment above.. I want to simplify/refactor this. I think I've devised a good way to do so, but changing the structure here a bit has some broad impact to the scope of this PR. I'd like to tackle this work promptly in a follow up PR.